Geometry/CommonTopologies/src/GEMStripTopology.cc

```diff
@@ -0,0 +1,125 @@
+/**GEMStripTopology
+ * based on CSCRadialStripTopology and TrapezoidalStripTopology
+ * \author Hyunyong Kim - TAMU
+ */
+#include "Geometry/CommonTopologies/interface/GEMStripTopology.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include <iostream>
+#include <cmath>
+#include <algorithm>
+
+GEMStripTopology::GEMStripTopology(int ns, float aw, float dh, float r0)
+    : numberOfStrips_(ns), angularWidth_(aw), detHeight_(dh), centreToIntersection_(r0) {
+  yAxisOrientation_ = 1;
+  phiOfOneEdge_ = -(0.5 * numberOfStrips_) * angularWidth_ * yAxisOrientation_;
+  yCentre_ = 0;
+  LogTrace("GEMStripTopology") << "Constructing GEMStripTopology with"
+                               << " nstrips = " << ns << " angular width = " << aw << " det. height = " << dh
+                               << " r0 = " << r0 << "\n";
+}
+
+GEMStripTopology::GEMStripTopology(int ns, float aw, float dh, float r0, float yAx)
+    : numberOfStrips_(ns), angularWidth_(aw), detHeight_(dh), centreToIntersection_(r0), yAxisOrientation_(yAx) {
+  phiOfOneEdge_ = -(0.5 * numberOfStrips_) * angularWidth_ * yAxisOrientation_;
+  yCentre_ = 0;
+  LogTrace("GEMStripTopology") << "Constructing GEMStripTopology with"
+                               << " nstrips = " << ns << " angular width = " << aw << " det. height = " << dh
+                               << " r0 = " << r0 << " yAxOrientation = " << yAx << "\n";
+}
+
+LocalPoint GEMStripTopology::localPosition(float strip) const {
+  return LocalPoint(yAxisOrientation() * originToIntersection() * tan(stripAngle(strip)), 0);
+}
+
+LocalPoint GEMStripTopology::localPosition(const MeasurementPoint& mp) const {
+  const float  // y = (L/cos(phi))*mp.y()*cos(phi)
+      y(mp.y() * detHeight() + yCentreOfStripPlane()),
+      x(yAxisOrientation() * yDistanceToIntersection(y) * std::tan(stripAngle(mp.x())));
+  return LocalPoint(x, y);
+}
+
+LocalError GEMStripTopology::localError(float strip, float stripErr2) const {
+  const double phi(stripAngle(strip)), t1(std::tan(phi)), t2(t1 * t1),
+      tt(stripErr2 * std::pow(centreToIntersection() * angularWidth(), 2)),  // tangential sigma^2   *c2
+      rr(std::pow(detHeight(), 2) * (1.f / 12.f)),  // radial sigma^2( uniform prob density along strip)  *c2
+
+      xx(tt + t2 * rr), yy(t2 * tt + rr), xy(t1 * (rr - tt));
+
+  return LocalError(xx, xy, yy);
+}
+
+LocalError GEMStripTopology::localError(const MeasurementPoint& mp, const MeasurementError& me) const {
+  const double phi(stripAngle(mp.x())), s1(std::sin(phi)), c1(std::cos(phi)), cs(s1 * c1), s2(s1 * s1),
+      c2(1 - s2),  // rotation matrix
+
```

Comment:
Please check (or assert) that `c1 != 0`.

Reply:
yes, I added the assert function.